integration/run-tests.sh

```diff
@@ -0,0 +1,92 @@
+
+test_ns="helmfile-tests"
+test_dir="./integration"
+helmfile="./helmfile --namespace=${test_ns}"
+helm="helm --kube-context=minikube"
+kubectl="kubectl --context=minikube --namespace=${test_ns}"
+
+# Helper Functions
+function fail () {
+    tput bold; tput setaf 1; echo -n "ERROR: "; tput sgr0; echo "${@}"
+    exit 1
+}
+function pass () {
+    tput bold; tput setaf 2; echo -n "PASS: "; tput sgr0; echo "${@}"
+}
+function info () {
+    tput bold; tput setaf 4; echo -n "INFO: "; tput sgr0; echo "${@}"
+}
+function test () {
+    tput bold; tput setaf 6; echo -n "TEST: "; tput sgr0; echo "${@}"
+}
+function warn () {
+    tput bold; tput setaf 3; echo -n "WARN: "; tput sgr0; echo "${@}"
+}
+function wait_deploy_ready() {
+    $kubectl rollout status deployment ${1}
+    while [ "$($kubectl get deploy ${1} -o=jsonpath='{.status.readyReplicas}')" == "0" ]; do
+        info "Waiting for deployment ${1} to be ready"
+        sleep 1
+    done
+}
+
+# Check environment is correctly setup
+
+if [ ! "$(basename $(pwd))" == "helmfile" ]; then
+    fail "Script needs to be run from the base directory of this repo."
+fi
+if ! hash minikube 2>/dev/null; then
+    fail "Minikube needs to be installed."
+fi
+if [ ! $(minikube status --format '{{.MinikubeStatus}}') == "Running" ]; then
+    fail "Minikube is not running."
+fi
+if [ ! $(minikube status --format '{{.ClusterStatus}}') == "Running" ]; then
+    fail "Minikube Cluster is not running."
+fi
+if ! kubectl version --short 1> /dev/null; then
+    fail "Could not connect to minikube apiserver"
+fi
+info "Connection to minikube successful"
+
+if ! hash curl 1>/dev/null; then
+    fail "curl needs to be installed."
+fi
+
+if ! hash docker 1>/dev/null; then
+    fail "Docker needs to be installed."
+fi
+if ! docker version 1> /dev/null; then
+    fail "Could not connect to Docker daemon"
+fi
+info "Connection to minikube docker daemon successful"
+
+if ! hash gox 1>/dev/null; then
+    fail "Gox needs to be installed. Run: make tools"
+fi
+
+if ! hash helm 1>/dev/null; then
+    fail "Helm needs to be installed."
+fi
+info "Using Helm version: $(helm version --short --client | grep -o v.*$)"
+
+
+info "Environment all set up, running tests"
+set -e
+$helm init 1> /dev/null
+make build 1> /dev/null
+$helmfile -v 1> /dev/null
+$kubectl get namespace ${test_ns} &> /dev/null && warn "Namespace ${test_ns} exists, from a previous test run?"
+
+test "happypath: rollout of httpbin chart"
+$helmfile -f ${test_dir}/happypath.yaml sync
+wait_deploy_ready httpbin-httpbin
+curl --fail $(minikube service --url --namespace=${test_ns} httpbin-httpbin)/status/200 || fail "httpbin failed to return 200 OK"
+$helmfile -f ${test_dir}/happypath.yaml delete
+$helm status --namespace=${test_ns} httpbin &> /dev/null && fail "release should not exist anymore after a delete"
+pass "happypath.yaml"
+
+# If we reach here, no command exited with a non-zero exit code and we passed all tests.
+$kubectl delete namespace ${test_ns}
```

Comment:
Almost nit but trap EXIT and run this cleanup inside the trap func would be a better approach because we then won't leave unnecessary resources when the script failed in the middle of process.

An example snippet(completely untested!):
```bash
on_exit() {
  $kubectl delete namespace ${test_ns}
}

trap 'on_exit' EXIT
```